src/Parser/DeathKnight/Blood/Modules/Talents/WillOfTheNecropolis.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SPELLS from 'common/SPELLS/index';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber } from "common/format";
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+
+const MINIMUMABSORTBTHRESHOLD = 0.05;
+
+class WillOfTheNecropolis extends Analyzer {
+
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  totalWotnAbsorbed = 0;
+  currentWotnAbsorbed = 0;
+  activated = 0;
+  spellDamageId = 0;
+  playerHealth = 0;
+  absorbToHeathPercent = 0;
+  goodAbsorbCount = 0;
+  nextEvent = false;
+
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTalent(SPELLS.WILL_OF_THE_NECROPOLIS_TALENT.id);
+  }
+
+  on_byPlayer_absorbed(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.WILL_OF_THE_NECROPOLIS_TALENT.id ) {
+      return;
+    }
+    this.totalWotnAbsorbed += event.amount;
+    this.currentWotnAbsorbed = event.amount;
+    this.activated += 1;
+    this.spellDamageId = event.extraAbility.guid;
+    this.nextEvent = true;
+  }
+
+  on_toPlayer_damage(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== this.spellDamageId || this.nextEvent === false) {
+      return;
+    }
+    this.nextEvent = false;
+    this.playerHealth = event.maxHitPoints;
+    this.absorbToHeathPercent = this.currentWotnAbsorbed / this.playerHealth;
```

Comment:
You can make this and playerHealth above local constants since they aren't used anywhere else, no need to define them as class variables. (also there's missing an L in absorbToHeathPercent)
```
const playerHealth = event.maxHitPoints;
const absorbToHeathPercent = this.currentWotnAbsorbed / playerHealth;
```